libraries/SPI/src/SPI.cpp

```diff
@@ -78,73 +94,100 @@ void SPIClass::usingInterrupt(uint8_t interruptNumber)
 
 void SPIClass::beginTransaction(uint8_t _pin, SPISettings settings)
 {
-  if(_pin > SPI_CHANNELS_NUM)
+  uint8_t idx;
+
+  if(_pin > NUM_DIGITAL_PINS)
     return;
 
-  spiSettings[_pin].clk = settings.clk;
-  spiSettings[_pin].dMode = settings.dMode;
-  spiSettings[_pin].bOrder = settings.bOrder;
-  if(spiSettings[_pin].bOrder == MSBFIRST) {
-    spiSettings[_pin].msb = MSBFIRST;
+  idx = getIdx(_pin);
+  if(idx == NB_SPI_SETTINGS) {
+    idx = addPin(_pin);
+    if(idx == NB_SPI_SETTINGS)
+      return;
```

Comment:
this case here is not clear - is this an error ?

Reply:
Yes this is an error because there is no more place available for a new CS pin.